src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -118,6 +120,7 @@ export class CellExecution {
     private disposables: IDisposable[] = [];
     private cancelHandled = false;
     private requestHandlerChain = Promise.resolve();
+    private activeExecutions: { execution: Promise<void>; session: IJupyterSession }[] = [];
```

Comment:
I've not played around with new native execution much, so this might just be a misunderstanding on my part (looking over it more now). But I thought that a new CellExecution was created each time we executed a notebook cell. So why would we be getting multiple active executions on one CellExecution?

Reply:
The way the code is structured it's possible? There's nothing that prevents (from the CellExecution class) there being multiple executions for the same cell. But the calling class does prevent that.

Reply:
I don't see how we can have more than one item in the array, if we do its incorrect or i'm mistaken.
Can we not have an array instead. As we create new instances of this class for each execution (even if we run the same cell multiple times).